src/zenml/pipelines/base_pipeline.py

```diff
@@ -432,107 +432,110 @@ def run(
         Returns:
             The result of the pipeline.
         """
-        if constants.SHOULD_PREVENT_PIPELINE_EXECUTION:
-            # An environment variable was set to stop the execution of
-            # pipelines. This is done to prevent execution of module-level
-            # pipeline.run() calls inside docker containers which should only
-            # run a single step.
-            logger.info(
-                "Preventing execution of pipeline '%s'. If this is not "
-                "intended behavior, make sure to unset the environment "
-                "variable '%s'.",
-                self.name,
-                constants.ENV_ZENML_PREVENT_PIPELINE_EXECUTION,
+        with event_handler(AnalyticsEvent.RUN_PIPELINE) as handler:
```

Comment:
I think this handler should only start after the if the checks if the pipeline gets executed.

Otherwise all our docker images that run steps and don't have `if __name__ == '__main__':` around the `pipeline.run()` call will log pipeline run events

Reply:
Nice catch. Fixing it!